ui/src/components/common/NotificationCenter.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+
+import Page from '../common/Page'
+import PageSection from './PageSection'
+import { useRelationshipRequestNotifications } from '../common/RelationshipRequestNotifications'
+import { useCCPCustomerNotifications } from '../CCP/CCPCustomerNotifications'
+import { useBrokerCustomerInviteNotifications } from '../Investor/BrokerCustomerInviteNotifications'
+import { useCCPCustomerInviteNotifications } from '../Investor/CCPInviteNotifications'
+import { useDismissibleNotifications } from '../common/DismissibleNotifications'
+import { useExchangeInviteNotifications } from '../Investor/ExchangeInviteNotifications'
+
+import { NotificationCenterIcon, ArrowLeftIcon } from '../../icons/Icons'
+
+type Props = {
+  sideNav: React.ReactElement;
+  onLogout: () => void;
+  showNotificationAlert?: boolean;
+  handleNotificationAlert?: () => void;
+}
+
+export const useAllNotifications = () => {
+  const notifications = [
+    ...useRelationshipRequestNotifications(),
+    ...useCCPCustomerNotifications(),
+    ...useCCPCustomerInviteNotifications(),
+    ...useBrokerCustomerInviteNotifications(),
+    ...useExchangeInviteNotifications(),
+    ...useDismissibleNotifications()
+  ];
+
+  return notifications;
+}
+
+const NotificationCenter: React.FC<Props> = ({ sideNav, onLogout, showNotificationAlert, handleNotificationAlert }) => {
+  const history = useHistory();
+  const notifications = useAllNotifications();
+  const [prevLocation, setPrevLocation] = useState<string>();
+  const [prevPath, setPrevPath] = useState<string>();
+
+  const historyState = history.location.state as string;
+
+  useEffect(() => {
+    const tempLocation = historyState?.split('/');
+    const tempPath = historyState;
+    if (historyState) {
+      localStorage.setItem('prevLocation', tempLocation[tempLocation.length - 1]);
+      localStorage.setItem('prevPath', tempPath);
+      setPrevLocation(tempLocation[tempLocation.length - 1]);
+      setPrevPath(tempPath);
+    } else {
+      const tempLocation = localStorage.getItem('prevLocation');
+      const tempPrevPath = localStorage.getItem('prevPath');
+      if (tempLocation) setPrevLocation(tempLocation);
+      if(tempPrevPath) setPrevPath(tempPrevPath);
+    }
+  });
+
+  const handleGoBack = () => {
+      history.push(`${prevPath}`);
```

Comment:
if prevPath is a string you probably dont need to wrap

Reply:
I tried but for some reason it dosn't like it like that... something about being a state. but it is a string...